### PR TITLE
home-environment: make sessionVariables a freeform module

### PR DESCRIPTION
**Pull request by rycee**
### Description

Also introduce explicit options for `PATH` and `NIX_PATH`.

### Checklist

- [X] Change is backwards compatible: Mostly, it does limit session variable values to `package`, `str`, `int`, and `float` but I think this covers pretty much all use-cases.

- [X] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all`.

- [X] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

### CODE REVIEW

File: modules/home-environment.nix [hunk 1]
```diff
@@ -229,7 +229,30 @@ in
 
     home.sessionVariables = mkOption {
       default = {};
-      type = types.attrs;
+      type = types.submodule {
+        freeformType = with types;
+          lazyAttrsOf (oneOf [ package str int float ]);
```
Comment (thiagokokada) [inline modules/home-environment.nix:234]: What is a `lazyAttrsOf` :thinking: ?

**Review by thiagokokada — APPROVED** [review]

File: modules/home-environment.nix (hunk 1 above)
Comment (berbiche) [reply modules/home-environment.nix:234]: If I understand correctly, `lazyAttrsOf` is used to delay the evaluation of the attribute set  to be used for conditional `optionalAttrs` and other such things where recursion would happen if `mkIf` and `mkMerge` were to be used.

**Comment by berbiche** [comment]
@rycee was there a consensus around what to do with environment variables in HM?

**Comment by rycee** [comment]
Not sure, I haven't had time to revisit this. Maybe in April.

**Comment by rycee** [comment]
Still planning to revisit this 🙂

**Comment by arcnmx** [comment]
@rycee should this have been closed? This has been a pain point for years, particularly that `sessionVariables` is a broken `attrs` type instead of `attrsOf`.

**Pull request closed without merging**